tensorflow_datasets/core/custom_dataset/image_folder_test.py

```diff
@@ -0,0 +1,115 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Test for ImageFolder."""
+
+import os
+
+import tensorflow_datasets.public_api as tfds
+
+_EXAMPLE_DIR = os.path.join(
+      tfds.testing.test_utils.fake_examples_dir(), 'image_folder_data')
+
+original_init = tfds.ImageFolder.__init__
+
+def new_init(self, root_dir=None, **kwargs):
+  assert root_dir is None
+  del kwargs
+  root_dir = _EXAMPLE_DIR
+  original_init(self, root_dir=root_dir)
+
+
+def download_and_prepare(self, **kwargs): # pylint:disable = unused-argument
+  pass
+
+
+class ImageFolderTest(tfds.testing.DatasetBuilderTestCase):
+  """Test for ImageFolder."""
+
+  DATASET_CLASS = tfds.ImageFolder
+  SPLITS = {
+      'train': 2,
+      'test': 6,
+  }
+  EXAMPLE_DIR = _EXAMPLE_DIR
+  
+  @classmethod
+  def setUpClass(cls): # pylint:disable = invalid-name
+    super(ImageFolderTest, cls).setUpClass()
+    cls.DATASET_CLASS.__init__ = new_init
```

Comment:
It would be cleaner to use mocking here.

Also add `tearDownClass` to stop the mock.

Reply:
See `DatasetBuilderTestCase` implementation for examples on how to use mock